api/buckets/buckets.go

```diff
@@ -106,23 +182,258 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.Threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	return b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
 }
 
 func (b *Buckets) Delete(ctx context.Context, dbID thread.ID, key string, opts ...Option) error {
 	args := &Options{}
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.Threads.Delete(ctx, dbID, CollectionName, []string{key}, db.WithTxnToken(args.Token))
+	return b.threads.Delete(ctx, dbID, CollectionName, []string{key}, db.WithTxnToken(args.Token))
+}
+
+// Archive pushes the current root Cid to the corresponding FFS instance of the bucket.
+// If it's not the first archive, we leverage the Replace() API to more efficient pinning.
+// The operation is async, so it will return as soon as the archiving is scheduled in Powergate.
+func (b *Buckets) Archive(ctx context.Context, dbID thread.ID, key string, c cid.Cid, opts ...Option) error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	ffsi, err := b.ffsCol.Get(ctx, key)
+	if err != nil {
+		return fmt.Errorf("getting ffs instance data: %s", err)
+	}
+
+	ctxFFS := context.WithValue(ctx, powc.AuthKey, ffsi.FFSToken)
+	var jid ffs.JobID
+	if ffsi.Archives.Current.JobID != "" {
+		oldCid, err := cid.Decode(ffsi.Archives.Current.Cid)
+		if err != nil {
+			return fmt.Errorf("parsing old Cid archive: %s", err)
+		}
+		if oldCid.Equals(c) {
+			return nil
+		}
+		jid, err = b.pgClient.FFS.Replace(ctxFFS, oldCid, c)
+		if err != nil {
+			return fmt.Errorf("replacing cid: %s", err)
+		}
+		ffsi.Archives.History = append(ffsi.Archives.History, ffsi.Archives.Current)
+	} else {
+		jid, err = b.pgClient.FFS.PushConfig(ctxFFS, c)
+		if err != nil {
+			return fmt.Errorf("pushing config: %s", err)
+		}
+	}
+
+	ffsi.Archives.Current = collections.Archive{
+		Cid:       c.String(),
+		CreatedAt: time.Now().Unix(),
+		JobID:     jid.String(),
+		JobStatus: int(ffs.Queued),
+	}
+	if err := b.ffsCol.Replace(ctx, ffsi); err != nil {
+		return fmt.Errorf("updating ffs instance data: %s", err)
+	}
+	go b.trackArchiveProgress(util.NewClonedContext(ctx), dbID, key, jid, ffsi.FFSToken, c, opts...)
+
+	return nil
+}
+
+// ArchiveStatus returns the last known archive status on Powergate. If the return status is Failed,
+// an extra string with the error message is provided.
+func (b *Buckets) ArchiveStatus(ctx context.Context, key string) (ffs.JobStatus, string, error) {
+	ffsi, err := b.ffsCol.Get(ctx, key)
+	if err != nil {
+		return ffs.Failed, "", fmt.Errorf("getting ffs instance data: %s", err)
+	}
+
+	if ffsi.Archives.Current.JobID == "" {
+		return ffs.Failed, "", ErrNoCurrentArchive
+	}
+	current := ffsi.Archives.Current
+	if current.Aborted {
```

Comment:
The `Aborted` indicates a situation in that the background goroutine ended unexpectedly and the Job status might be in an indefinite state. This should be re-handled eventually, or similar.
The summary is: if it's aborted, we're not sure how the Job ended. Doesn't necessarily means it failed, just that maybe the network conn exploded and we lost track of the status.

It's important eventually to add further code to recover these cases, etc.